packages/report-server/src/__tests__/reportBuilder/query/QueryBuilder.test.ts

```diff
@@ -82,23 +82,33 @@ describe('QueryBuilder', () => {
   const outputQuery = (input: Partial<FetchReportQuery> = {}) => ({
     hierarchy: HIERARCHY,
     organisationUnitCodes: ['TO'],
-    period: getPeriodString([2017, 1], [2020, 12]),
+    period: getPeriodString([2017, 1], [2020, 12], 15),
     startDate: '2017-01-01',
-    endDate: '2020-12-31',
+    endDate: '2020-12-15',
     ...input,
   });
 
   const getPeriodString = (
     [startYear, startMonth]: [number, number],
     [endYear, endMonth]: [number, number],
+    endDay?: number,
+    startDay?: number,
   ) => {
     const periods = [];
 
     for (let year = startYear; year <= endYear; year++) {
       const currentStartMonth = year === startYear ? startMonth : 1;
       const currentEndMonth = year === endYear ? endMonth : 12;
       for (let month = currentStartMonth; month <= currentEndMonth; month++) {
-        periods.push(`${year}${month.toString().padStart(2, '0')}`);
+        if (month === currentEndMonth && year === endYear && endDay) {
+          for (let day = startDay || 1; day <= endDay; day++) {
```

Comment:
Just be a tad careful using the `||` operator here. If `startDay` were `0` then it'd still result it in being `1` here. While that works fine in this specific code so I guess its fine it is always a bit surprising 😮 Falsey behaviour is a bit risky